File: Problems/1299.replace-elements-with-greatest-element-on-right-side.py
#
# @lc app=leetcode id=1299 lang=python3
#
# [1299] Replace Elements with Greatest Element on Right Side
#

# @lc code=start
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        # Keep track of current maximum
        currentMax = -1
        # Loop backwards
        for i in range(len(arr)-1, 0, -1):
            if arr[i] > currentMax:
                currentMax = arr[i]
            arr[i] = currentMax
        # Shift array over by one, then set the last value
        arr[:-1] = arr[1:]
        arr[-1] = -1
        return arr

# @lc code=end

File: leetcode/Problems/1299.replace-elements-with-greatest-element-on-right-side.java
/*
 * @lc app=leetcode id=1299 lang=java
 *
 * [1299] Replace Elements with Greatest Element on Right Side
 */

// @lc code=start
class Solution {
    public int[] replaceElements(int[] arr) {
        int currentMax = -1;
        for (int i = arr.length-1; i >= 0; i--) {
            int arrTemp = arr[i];
            arr[i] = currentMax;
            if (arrTemp > currentMax) currentMax = arrTemp;
        }
        return arr;
    }
}
// @lc code=end

